builtin/mainmenu/dlg_csm.lua

```diff
@@ -0,0 +1,193 @@
+--Luanti
+--Copyright (C) 2025 ProunceDev <prouncedev@gmail.com>
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local packages_raw, packages, tab_selected_pkg
+
+local function modname_valid(name)
+	return not name:find("[^a-z0-9_]")
+end
+
+local function get_formspec(dlgview, name, tabdata)
+	if pkgmgr.clientmods == nil then
+		pkgmgr.reload_global_mods()
+	end
+
+	if packages == nil then
+		packages_raw = {}
+		table.insert_all(packages_raw, pkgmgr.clientmods:get_list())
+
+		local function get_data()
+			return packages_raw
+		end
+
+		local function is_equal(element, uid) --uid match
+			return (element.type == "game" and element.id == uid) or
+					element.name == uid
+		end
+
+		packages = filterlist.create(get_data, pkgmgr.compare_package, is_equal, nil, {})
+
+		local filename = core.get_clientmodpath() .. DIR_DELIM .. "mods.conf"
+		local conffile = Settings(filename)
+		local mods = conffile:to_table()
+
+		for i = 1, #packages_raw do
+			local mod = packages_raw[i]
+			if mod.is_clientside then
+				if modname_valid(mod.name) then
+					conffile:set("load_mod_" .. mod.name,
+						mod.enabled and "true" or "false")
+				elseif mod.enabled then
+					gamedata.errormessage = fgettext_ne("Failed to enable clientmod" ..
+						" \"$1\" as it contains disallowed characters. " ..
+						"Only characters [a-z0-9_] are allowed.",
+						mod.name)
+				end
+				mods["load_mod_" .. mod.name] = nil
```

Comment:
Duplicate of `dlg_config_world.lua`. Please re-use existing code. This dialogue already mentions that it's about Client Mods, thus the error message (`Failed to enable mod [...]` (that is used there) should be good enough.

Example:
```lua
function pkgmgr.foobar_insert_my_mods(mods_ret, packages, cb_may_add)
	for _, mod in ipairs(packages) do
		if cb_may_add(mod) then
			-- same code here
		end
	end
end


-- for CSM:
pkgmgr.foobar_insert_my_mods(mods, packages_raw, function(mod)
	return mod.is_clientside
end

-- for Server Mods: (dlg_config_world.lua)
pkgmgr.foobar_insert_my_mods(mods, packages_raw, function(mod)
	return not mod.is_modpack and not mod.is_game_content
end
```